cowprotocol/cowswap
# fix(permit): improve permittable detection

# Summary

Fixes #3221

WC network switching was not changing the provider network

# To Test

1. Delete the `permittableTokens` localStorage key if you have it
2. Connect via WCv2 to Goerli or Gchain
3. Filter the console logs by `permit`
4. Switch between different tokens
* On gchain none will be permittable and you'll see something like: 
    `[checkTokenIsPermittable] Not a permittable token 0x91056d4a53e1faa1a84306d4deaec71085394bc8 nonce not supported`
* On Goerli you should see the above only for unsupported. Supported should not print anything. You can check the `permittableTokens` localStorage key to see the token info
5. Switch to mainnet and try different tokens
* Same result as on Goerli. Permittable tokens will not log anything, only visible in the localStorage
